### PR TITLE
LibJS: Add helpers to Cell::Visitor for visiting AK collections

### PR DESCRIPTION
**Pull request by awesomekling**
...and use them to simplify visit functions everywhere.

### CODE REVIEW

File: Userland/Libraries/LibJS/Heap/Cell.h [hunk 1]
```diff
@@ -73,6 +74,63 @@ class Cell {
...
+            for (auto& value : span)
+                visit(value);
+        }
+
+        template<typename T>
+        void visit(Vector<T> const& vector)
+        {
+            for (auto& value : vector)
+                visit(value);
+        }
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Heap/Cell.h:96]: I'd suspect we don't need this one? The span overloads should cover Vector, like they're doing for e.g. Array

File: Userland/Libraries/LibJS/Heap/Cell.h [hunk 2]
```diff
@@ -73,6 +74,63 @@ class Cell {
...
+
+        template<typename K, typename V, typename T>
+        void visit(OrderedHashMap<K, V, T> const& map)
+        {
+            for (auto& [key, value] : map) {
+                if constexpr (IsBaseOf<Cell, K>)
+                    visit(key);
+                if constexpr (IsBaseOf<Cell, V>)
+                    visit(value);
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Heap/Cell.h:130]: Does this work for e.g. `HashMap<int, GCPtr<something>>` or `HashMap<int, Value>`? In those cases, I don't think we'll pass this check.

Maybe we can get away with a requires clause, something like (untested):
```c++
if constexpr (requires { visit(value); })
    visit(value);
````

**Review by trflynn89 — COMMENTED** [review]
I can't believe we didn't do this before, this is so much better

File: Userland/Libraries/LibJS/Heap/Cell.h (hunk 2 above)
Comment (awesomekling) [reply Userland/Libraries/LibJS/Heap/Cell.h:130]: Yeah this is indeed busted the way I did it. Let's do a `requires` :)

File: Userland/Libraries/LibJS/Heap/Cell.h (hunk 1 above)
Comment (awesomekling) [reply Userland/Libraries/LibJS/Heap/Cell.h:96]: Alas it seems we do indeed need it

**Pull request merged**